overlord/devicestate/handlers_install.go

```diff
@@ -205,11 +208,11 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 	var installedSystem *install.InstalledSystemSideData
 	// run the create partition code
 	logger.Noticef("create and deploy partitions")
-	func() {
+	timings.Run(perfTimings, "install-run", "Install the run system", func(timings.Measurer) {
 		st.Unlock()
 		defer st.Lock()
-		installedSystem, err = installRun(model, gadgetDir, kernelDir, "", bopts, installObserver)
-	}()
+		installedSystem, err = installRun(model, gadgetDir, kernelDir, "", bopts, installObserver, perfTimings)
```

Comment:
Did you mean to nest timings here? If so, this should pass timings.Measurer that the lambda receives, not perfTimings.

Reply:
Nice catch!

Reply:
With this the timings look more sensible now:
```
ID   Status        Doing      Undoing  Summary
58   Doing             -            -  Ensure next boot to run mode
57   Done         3589ms            -  Setup system for run mode
 ^                1193ms            -    Install the run system
  ^                639ms            -      Create partitions
  ^                124ms            -      Create filesystem for system-boot
  ^                104ms            -      Write content for system-boot
  ^                 20ms            -      Create filesystem for system-save
  ^                  8ms            -      Write content for system-save
  ^                 44ms            -      Create filesystem for system-data
 ^                2379ms            -    Make target system runnable
```